lesson22/app.js

```diff
@@ -1,225 +1,235 @@
-const url = "https://jsondata.okiba.me/v1/json/tN7Ay210317141032";
 const parent = document.getElementById("js-parent");
-const table = document.createElement("table");
 
-table.id = "table";
-table.classList.add("table");
+const createAttributedElements = (tag, attrObj, str) => {
+  const element = document.createElement(tag);
+  Object.keys(attrObj).forEach((attribute) => {
+    element.setAttribute(attribute, attrObj[attribute]);
+  });
+  if (str !== undefined) element.textContent = str;
+  return element;
+};
 
 const wait = (sec) => {
   return new Promise((resolve) => {
     setTimeout(resolve, sec * 1000);
   });
 };
 
-async function getJsonData() {
+const getData = async () => {
   try {
     await wait(3);
-    const response = await fetch(url);
-    const jsonData = await response.json();
-    return jsonData;
-  } catch (err) {
-    throw Error(err);
-  } finally {
-    console.log("終了しました");
+    const response = await fetch(
+      "https://mocki.io/v1/d80d85fd-fc83-4c8f-8f05-e393a4cd5508"
+    );
+    if (!response.ok)
+      throw new Error(`${response.status}:${response.statusText}`);
+    return await response.json();
+  } catch (error) {
+    renderErrorMessage(error);
+    console.error(error);
   }
-}
+};
+
+const renderLoading = () => {
+  const loadingImage = document.createElement("span");
+  loadingImage.classList.add("animation-loader");
+  parent.appendChild(loadingImage);
+};
 
-async function init() {
+const removeLoading = () =>
+  document.querySelector(".animation-loader").remove();
+
+const renderErrorMessage = (error) => {
+  const errorMessage = document.createElement("p");
+  errorMessage.textContent = error;
+  parent.appendChild(errorMessage);
+};
+
+const init = async () => {
+  renderLoading();
   let data;
   try {
-    data = await getJsonData();
-  } catch (e) {
-    console.error();
+    data = await getData();
+  } catch (error) {
+    renderErrorMessage(error);
+    console.error(error);
+    return;
   } finally {
-    console.log("終了しました");
+    removeLoading();
   }
-  if (data.length !== 0) {
-    createElements(data);
-  } else {
-    parent.innerHTML = "data is empty";
-  }
-}
+  if (data.data.length === 0) return (parent.textContent = "data is empty");
```

Comment:
nits
```suggestion
  if (hasNoData(data.data.length)) return (parent.textContent = "data is empty");
```
と言うふうにすれば少し意図が伝わり、読みやすく&どこでも使えます

Reply:
ありがとうございます！
hasNoData()を新しく作成・追加しました。
[1130594](https://github.com/443n0511/js-lesson/pull/44/commits/113059446182b0d74b0a1c7ebbd068cfe996f51b)